lib/foreman/cast.rb

```diff
@@ -0,0 +1,26 @@
+module Foreman
+  module Cast
+    def self.to_bool(value)
+      case value
+
+      when String
+        return true if value =~ (/\A(true|t|yes|y|on|1)\z/i)
+        return false if value.blank? || value =~ (/\A(false|f|no|n|off|0)\z/i)
```

Comment:
> How is an empty object "effectively nil"?

The `["setting"]["value"]` parameter is nil.

In this case I think the bug is with the controller, which isn't checking for presence before attempting to parse and cast the value, and not to_bool.